cudax/test/memory_resource/async_memory_pool.cu

```diff
@@ -209,3 +212,258 @@ TEST_CASE("async_memory_pool comparison", "[memory_resource]")
     CHECK(current_default_pool != first);
   }
 }
+
+TEST_CASE("async_memory_pool accessors", "[memory_resource]")
+{
+  int current_device{};
+  {
+    _CCCL_TRY_CUDA_API(::cudaGetDevice, "Failed to querry current device with with cudaGetDevice.", &current_device);
+  }
+
+  int driver_version = 0;
+  {
+    _CCCL_TRY_CUDA_API(::cudaDriverGetVersion, "Failed to call cudaDriverGetVersion", &driver_version);
+  }
+
+  ::cudaMemPool_t current_default_pool{};
+  {
+    _CCCL_TRY_CUDA_API(::cudaDeviceGetDefaultMemPool,
+                       "Failed to call cudaDeviceGetDefaultMemPool",
+                       &current_default_pool,
+                       current_device);
+  }
+
+  SECTION("async_memory_pool::set_attribute")
+  {
+    cudax::mr::async_memory_pool pool{current_device};
+
+    { // cudaMemPoolReuseFollowEventDependencies
```

Comment:
Could this repeated test be a template (even a lambda)?

Reply:
The issue is that some attributes do not have setters, some need an allocation some dont